libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt

```diff
@@ -1273,17 +1273,40 @@ object ArrayOps : TemplateGroupBase() {
     }
 
     val f_fill = fn("fill(element: T, fromIndex: Int = 0, toIndex: Int = size)") {
-        platforms(Platform.JVM)
         include(InvariantArraysOfObjects, ArraysOfPrimitives, ArraysOfUnsigned)
     } builder {
         doc { "Fills original array with the provided value." }
         returns("Unit")
         body {
             """
-            java.util.Arrays.fill(this, fromIndex, toIndex, element)
+            if (fromIndex !in 0..size) {
+              throw IndexOutOfBoundsException("fromIndex ${'$'}fromIndex out of range [0, ${'$'}size]")
+            }
+            if (toIndex > size) {
+              throw IndexOutOfBoundsException("toIndex ${'$'}toIndex out of range [${'$'}fromIndex, ${'$'}size]")
+            }
+            require(fromIndex <= toIndex) { "fromIndex(${'$'}fromIndex) > toIndex(${'$'}toIndex)" }
+            for (index in fromIndex until toIndex) {
+                this[index] = element
+            }
             """
         }
 
+        on(Platform.JVM) {
+            suppress("ACTUAL_FUNCTION_WITH_DEFAULT_ARGUMENTS")
+            body {
+                """
+                java.util.Arrays.fill(this, fromIndex, toIndex, element)
+                """
+            }
+        }
+        on(Platform.JS) {
```

Comment:
JS has https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill but not sure it's worth delegating to or not.

Reply:
I think it's definitely worth it, however we should still check argument values to enforce our contract of this method.

Reply:
Ok.

 > we should still check argument values to enforce our contract of this method

Yes JS seems to just silently ignore out-of-bounds and to < from 😑

> We'll need to provide polyfills though, as we target ES 5 that doesn't have these methods in the standard.

Do you know of a similar API does this that I can copy? I can go searching for one, but thought you might know one off-hand.

Reply:
The MDN article you have linked contains a polyfill implementation. We keep 'em collection of polyfills used in stdlib here: https://github.com/JetBrains/kotlin/blob/1.3.50/libraries/stdlib/js-v1/src/js/polyfills.js

Reply:
Updated. PTAL!